contrib/ci/qt5-thread-test/qt-thread-test.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2020 Aleix Pol <aleixpol@kde.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+extern "C" {
+#include <fwupd.h>
+}
+
+#include <QCoreApplication>
+#include <QtConcurrentRun>
+#include <QFutureWatcher>
+
+int main(int argc, char** argv)
+{
+    QCoreApplication app(argc, argv);
+
+    auto client = fwupd_client_new();
+    auto cancellable = g_cancellable_new();
+    g_autoptr(GError) error = nullptr;
+
+    auto fw = new QFutureWatcher<GPtrArray*>(&app);
+    QObject::connect(fw, &QFutureWatcher<GPtrArray*>::finished, [fw]() {
+        QCoreApplication::exit(0);
+    });
+    fw->setFuture(QtConcurrent::run([&] {
+        return fwupd_client_get_devices(client, cancellable, &error);
+    }));
```

Comment:
if this hangs, how do we signify a CI failure? Is github actions going to kill our build after say 30 minutes? I'd much rather have a timer somewhere.

Reply:
There is a timeout set on the test in meson, so a hang should be killed by the test runner I would expect